index.php

```diff
@@ -320,54 +322,64 @@
-//			$func = str_replace('-','_',$a->theme_info["extends"]) . '_init';
-//			$func($a);
-//		}
-//	}
-
-	if(($_SERVER['REQUEST_METHOD'] === 'POST') && (! $a->error)
-		&& (function_exists($a->module . '_post'))
-		&& (! x($_POST,'auth-params'))) {
-		call_hooks($a->module . '_mod_post', $_POST);
-		$func = $a->module . '_post';
-		$func($a);
-	}
+		if(function_exists(str_replace('-','_',current_theme()) . '_init')) {
+			$func = str_replace('-','_',current_theme()) . '_init';
+			$func($a);
+		}
+	//	elseif (x($a->theme_info,"extends") && file_exists("view/theme/".$a->theme_info["extends"]."/theme.php")) {
+	//		require_once("view/theme/".$a->theme_info["extends"]."/theme.php");
+	//		if(function_exists(str_replace('-','_',$a->theme_info["extends"]) . '_init')) {
+	//			$func = str_replace('-','_',$a->theme_info["extends"]) . '_init';
+	//			$func($a);
+	//		}
+	//	}
+
+		if(($_SERVER['REQUEST_METHOD'] === 'POST') && (! $a->error)
+			&& (function_exists($a->module . '_post'))
+			&& (! x($_POST,'auth-params'))) {
+			call_hooks($a->module . '_mod_post', $_POST);
+			$func = $a->module . '_post';
+			$func($a);
+		}
 
-	if((! $a->error) && (function_exists($a->module . '_afterpost'))) {
-		call_hooks($a->module . '_mod_afterpost',$placeholder);
-		$func = $a->module . '_afterpost';
-		$func($a);
-	}
+		if((! $a->error) && (function_exists($a->module . '_afterpost'))) {
+			call_hooks($a->module . '_mod_afterpost',$placeholder);
+			$func = $a->module . '_afterpost';
+			$func($a);
+		}
 
-	if((! $a->error) && (function_exists($a->module . '_content'))) {
-		$arr = array('content' => $a->page['content']);
-		call_hooks($a->module . '_mod_content', $arr);
-		$a->page['content'] = $arr['content'];
-		$func = $a->module . '_content';
-		$arr = array('content' => $func($a));
-		call_hooks($a->module . '_mod_aftercontent', $arr);
-		$a->page['content'] .= $arr['content'];
-	}
+		if((! $a->error) && (function_exists($a->module . '_content'))) {
+			$arr = array('content' => $a->page['content']);
+			call_hooks($a->module . '_mod_content', $arr);
+			$a->page['content'] = $arr['content'];
+			$func = $a->module . '_content';
+			$arr = array('content' => $func($a));
+			call_hooks($a->module . '_mod_aftercontent', $arr);
+			$a->page['content'] .= $arr['content'];
+		}
 
-	if(function_exists(str_replace('-','_',current_theme()) . '_content_loaded')) {
-		$func = str_replace('-','_',current_theme()) . '_content_loaded';
-		$func($a);
+		if(function_exists(str_replace('-','_',current_theme()) . '_content_loaded')) {
+			$func = str_replace('-','_',current_theme()) . '_content_loaded';
+			$func($a);
+		}
+	} catch(HTTPException $e) {
+		logger('index.php: HTTPException '. $e->httpcode . ' : '. $_SERVER['REQUEST_URI'] . ' ADDRESS: ' . $_SERVER['REMOTE_ADDR'] . ' QUERY: ' . $_SERVER['QUERY_STRING'], LOGGER_DEBUG);
+		$e->send_header();
+		//TODO: template for main http response (404, 500,...)
```

Comment:
STANDARDS: please use three slashes and `@` so it will be included in doxygen - `/// @todo: template....`